File: Problems_and_Solutions/days-10 to 19/day-12/Java/Solution2.java
import java.util.Arrays;

class Solution2 {
    public static int[] searchRange(int[] nums, int target) {
        int n = nums.length;
        int first = first(nums, target, n);
        int last = last(nums, target, n);
        if (first == n || nums[first] != target) {
            return new int[]{-1, -1};
        }
        return new int[]{first, last - 1};
    }

    public static int first(int[] nums, int target, int n) {
        int low = 0, high = n - 1, ans = n;
        while (low <= high) {
            int mid = (low + high) / 2;
            if (nums[mid] >= target) {
                ans = mid;
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        return ans;
    }

    public static int last(int[] nums, int target, int n) {
        int low = 0, high = n - 1, ans = n;
        while (low <= high) {
            int mid = (low + high) / 2;
            if (nums[mid] > target) {
                ans = mid;
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        return ans;
    }

    public static void main(String[] args) {
        int[] nums = {5, 7, 7, 8, 8, 10};
        int target = 8;
        int[] result = searchRange(nums, target);
        System.out.println(Arrays.toString(result));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-12/c++/solution 2.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int length=nums.size();
        if(length==0)
            return vector<int>{-1,-1};
        int low1=0,low2=0;
        int high1=length-1;
        int high2=length-1;
        int mid=0;
        while(low1<=high1)
        {
            mid=(low1+high1)/2;
            if(target<=nums[mid])
                high1=mid-1;
            else
                low1=mid+1;
        }
        while(low2<=high2)
        {
            mid=(low2+high2)/2;
            if(target>=nums[mid])
                low2=mid+1;
            else
                high2=mid-1;
        }
        low1=low1>nums.size()-1 || nums[low1]!=target?-1:low1;
        high2=high2<0 || nums[high2]!=target?-1:high2;
        return vector<int>{low1,high2};
    }
};